### DISCUSSION TITLE
Accessing "home" inside darwing and vice-versa

### DISCUSSION THREAD

**Discussion by proxysoul** [juspay/nix-dev-home#95] [Q&A] [answered]
Hi all, im new to nix (few days old, so far enjoying it and great community) 
one thing i would like to achieve is to be able to access home-manager inside darwin and vice-versa 

any suggestions or simple way
using nix-darwin !

here is the [**Config**](https://github.com/Pouiiro/PouiiT-Files/tree/nixified)

Thanks a lot :)

EDIT: added config link

**Comment by srid** [comment]
Welcome, @Pouiiro 

> one thing i would like to achieve is to be able to access home-manager inside darwin and vice-versa

Can you elaborate on what you are trying to achieve? You are already accessing home-manager inside of your darwin configuration, here:

https://github.com/Pouiiro/PouiiT-Files/blob/4b09565dd59662af07da88809af8bed0631aa175/configurations/darwin/Mac.nix#L18-L21

**Reply by proxysoul** [reply]
i meant for example from here: 
https://github.com/Pouiiro/PouiiT-Files/blob/4b09565dd59662af07da88809af8bed0631aa175/modules/darwin/services/sketchybar.nix
 
now i have a separate configuration for home here
 https://github.com/Pouiiro/PouiiT-Files/blob/4b09565dd59662af07da88809af8bed0631aa175/modules/home/sketchybar-home.nix
 
but i would like to just access home.file from inside modules/darwin/services/sketchybar.nix 

i will test again and see, ill report the error i'm getting in a sec

**Reply by srid** [reply, marked as answer]
Both those modules can take a `flake` argument, that has access to the whole flake. For example,

https://github.com/srid/nixos-config/blob/a420e5f531172aef753b07a411de8e254207f5c6/modules/darwin/default.nix#L2-L5

This is actually mentioned here: https://nixos-unified.org/#why

<img width="524" alt="image" src="https://github.com/user-attachments/assets/fa0b0178-c34e-44c8-8f6d-94402c4f4897">

But we should document it more thoroughly.

Let me know if this addresses your use case.

**Reply by proxysoul** [reply]
Aah oki now i understand! thanks i got it to work!

**Reply by srid** [reply]
Now documented here: https://nixos-unified.org/specialArgs.html

Note that this pattern is already used in the template:

<img width="310" alt="image" src="https://github.com/user-attachments/assets/40490b46-bc74-4c99-b0ea-ff60dfbf2085">
